packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -1085,12 +1212,22 @@ export class AccountHandler {
 
     const createKeyFetchToken = async () => {
       if (requestHelper.wantsKeys(request)) {
-        keyFetchToken = await this.signinUtils.createKeyFetchToken(
-          request,
-          accountRecord,
-          password,
-          sessionToken
-        );
+        if (password.clientVersion === 2) {
+          keyFetchToken2 = await this.signinUtils.createKeyFetchToken(
+            request,
+            accountRecord,
+            password,
+            sessionToken
+          )
+        }
+        else {
+          keyFetchToken = await this.signinUtils.createKeyFetchToken(
+            request,
+            accountRecord,
+            password,
+            sessionToken
+          );
+        }
```

Comment:
I'm a bit concerned how this gonna work out.  Firefox still uses `keyFetchToken` in some places, so it being missing from the account data seems risky.

Reply:
The reason for the keyStretchToken2 convention, is to avoid unwrapping a keyFetchToken that was generated for a V2 password with a V1 password. This of course would result in the wrong kB. It seems like it's better for things to fail hard and in an obvious way, than to potentially produce a kB that is incorrect.

As long as the client authenticates with a V1 key stretch, then they get back a keyFetchToken as always. In my mind some of this is being kicked down the road until we do the client side integration. As it currently stands this change isn't incompatible because no V2 passwords are being created. It's definitely an excellent call out, and something we need to consider when making the front end changes and deciding whether a client authenticates with V1 or V2 key stretching.  

It's possible that RP who use web flows to log in, but derive kB themselves will have to provide an extra query parameter so we know they are ready to use V2 key stretching. I'll also need to go dig in search fox a bit more and see how this works.

Reply:
> As long as the client authenticates with a V1 key stretch, then they get back a keyFetchToken as always. ...

The tricky bit is that the FxA frontend will be the client and then hand the account data to Firefox, so we'd have a v2 client but a long tail of desktop Firefox installs that still might use the keyFetchToken.

> In my mind some of this is being kicked down the road until we do the front end portion.

Sure, it's not something we need to handle in this PR.

Reply:
I've filed FXA-9003 to help address this point. After digging into the FxA module in Firefox, it becomes apparent that V1 client side key stretching is being used internally to compute authPW and unwrapBKey. So this code must be updated anyways. 

As always, thanks for bringing this up! We wouldn't want to drop the ball here, and this is a requirement that certainly impacts the way the front-end changes will be made.